### PR TITLE
fix: and refactor root writer closing

### PR DESCRIPTION
**Pull request by andiwand**
align the closing mechanism of the root writers

some writer did not close the file which will result in empty files before the destructor is called

### CODE REVIEW

File: Examples/Io/Root/src/RootSimHitWriter.cpp [hunk 1]
```diff
@@ -70,23 +70,17 @@ ActsExamples::RootSimHitWriter::~RootSimHitWriter() {
...
-    m_outputTree->Write();
-    ACTS_VERBOSE("Wrote hits to tree '" << m_cfg.treeName << "' in '"
-                                        << m_cfg.filePath << "'");
-    m_outputFile->Close();
-  }
+  m_outputFile->cd();
+  m_outputTree->Write();
+  ACTS_VERBOSE("Wrote hits to tree '" << m_cfg.treeName << "' in '"
+                                      << m_cfg.filePath << "'");
+  m_outputFile->Close();
```
Comment (AJPfleger) [inline Examples/Io/Root/src/RootSimHitWriter.cpp:77]:
```suggestion
  m_outputFile->cd();
  m_outputTree->Write();
  m_outputFile->Close();
  ACTS_VERBOSE("Wrote hits to tree '" << m_cfg.treeName << "' in '"
                                      << m_cfg.filePath << "'");
```

We have here and at the other positions different orderings of `ACTS_VERBOSE` and `->Close()`. Should we decide for one?

**Review by AJPfleger — CHANGES_REQUESTED** [review]

File: Examples/Io/Root/src/RootSimHitWriter.cpp (hunk 1 above)
Comment (andiwand) [reply Examples/Io/Root/src/RootSimHitWriter.cpp]: good idea :+1:

File: Examples/Io/Root/src/RootTrackParameterWriter.cpp [hunk 2]
```diff
@@ -104,13 +104,12 @@ ActsExamples::RootTrackParameterWriter::~RootTrackParameterWriter() {
...
-    m_outputTree->Write();
-    ACTS_INFO("Write estimated parameters from seed to tree '"
-              << m_cfg.treeName << "' in '" << m_cfg.filePath << "'");
-    m_outputFile->Close();
-  }
+  m_outputFile->cd();
+  m_outputTree->Write();
+  ACTS_INFO("Write estimated parameters from seed to tree '"
+            << m_cfg.treeName << "' in '" << m_cfg.filePath << "'");
+  m_outputFile->Close();
```
Comment (AJPfleger) [inline Examples/Io/Root/src/RootTrackParameterWriter.cpp:111]:
```suggestion
  m_outputFile->cd();
  m_outputTree->Write();
  m_outputFile->Close();
  
    ACTS_INFO("Wrote estimated parameters from seed to tree '"
            << m_cfg.treeName << "' in '" << m_cfg.filePath << "'");
```

File: Examples/Io/Root/src/RootTrajectoryStatesWriter.cpp [hunk 3]
```diff
@@ -219,18 +219,17 @@ ActsExamples::RootTrajectoryStatesWriter::RootTrajectoryStatesWriter(
...
-    m_outputFile->cd();
-    m_outputTree->Write();
-    ACTS_INFO("Write states of trajectories to tree '"
-              << m_cfg.treeName << "' in '" << m_cfg.treeName << "'");
-  }
+  m_outputFile->cd();
+  m_outputTree->Write();
+  m_outputFile->Close();
+
+  ACTS_INFO("Write states of trajectories to tree '"
```
Comment (AJPfleger) [inline Examples/Io/Root/src/RootTrajectoryStatesWriter.cpp:230]:
```suggestion
  ACTS_INFO("Wrote states of trajectories to tree '"
```

File: Examples/Io/Root/src/RootTrajectorySummaryWriter.cpp [hunk 4]
```diff
@@ -129,29 +129,24 @@ ActsExamples::RootTrajectorySummaryWriter::RootTrajectorySummaryWriter(
...
-    m_outputTree->Write();
-    ACTS_INFO("Write parameters of trajectories to tree '"
-              << m_cfg.treeName << "' in '" << m_cfg.filePath << "'");
-  }
+  m_outputFile->cd();
+  m_outputTree->Write();
+  m_outputFile->Close();
+
+  ACTS_INFO("Write parameters of trajectories to tree '"
+            << m_cfg.treeName << "' in '" << m_cfg.filePath << "'");
```
Comment (AJPfleger) [inline Examples/Io/Root/src/RootTrajectorySummaryWriter.cpp:141]:
```suggestion
  ACTS_INFO("Wrote parameters of trajectories to tree '"
            << m_cfg.treeName << "' in '" << m_cfg.filePath << "'");
```

**Comment by andiwand** [comment]
should we get this in @AJPfleger ?

**Review by AJPfleger — APPROVED** [review]
lgtm

**Pull request merged**